include/inviwo/qt/editor/processormimedata.h

```diff
@@ -46,32 +46,22 @@ class IVW_QTEDITOR_API ProcessorMimeData : public QMimeData {
     Q_OBJECT
 #include <warn/pop>
 public:
-    ProcessorMimeData(std::unique_ptr<Processor> processor);
+    ProcessorMimeData(std::shared_ptr<Processor> processor);
     virtual ~ProcessorMimeData() = default;
 
-    std::unique_ptr<Processor> get() const;
+    std::shared_ptr<Processor> get() const;
     Processor* processor() const;
 
     static const std::string& getMimeTag();
 
     static const ProcessorMimeData* toProcessorMimeData(const QMimeData* data);
 
 private:
-    struct ContextAwareDelete {
-        template <typename T>
-        ContextAwareDelete(std::default_delete<T>&&) noexcept {}
+    std::shared_ptr<Processor> processor_;
 
-        template <typename T>
-        operator std::default_delete<T>() {
-            return {};
-        }
-
-        void operator()(Processor* p) const noexcept;
-    };
-
-    mutable std::unique_ptr<Processor, ContextAwareDelete> processor_;
+    // this guard will run on destruction before deleting the processor if we have the last
```

Comment:
I think that this is what is meant
```suggestion
    // This guard will activate the default context before deleting the processor. 
    // The guard will only run if we have the last processor reference. 
```

Reply:
updated the comment a bit